session.go

```diff
@@ -406,6 +420,7 @@ func (s *Session) send() {
 					if err != nil {
 						s.logger.Printf("[ERR] yamux: Failed to write header: %v", err)
 						asyncSendErr(ready.Err, err)
+						s.sendDeadCh <- true
```

Comment:
I don't think it can be done with defer, because it's important that the `sendDeadCh` has been signaled before entering `s.exitErr(err)`.

Reply:
Well, if it's close, deferring the close it in addition to explicit close where needed should be fine.

Reply:
Can you enlighten me about why sendDeadCh needs to complete the send before `s.exitErr(err)`? I don't see it yet

Reply:
Well now it's `close(s.sendDeadCh)`. It needs to be done first because `s.exitErr(err)` will itself do `Close` which waits for `s.sendDeadCh` to become closed, and only the sending goroutine can close it.